### PR TITLE
add ip generate & synthesis

### PR DESCRIPTION
**Pull request by kazutoiris**
1. Add ip generate & synthesis: Now, the script will automatically source `$(SRCDIR)/*.tcl` to complete IP generation and synthesis.

2. Move part definitions external: Part can be defined in the `Makefile`.

3. Disable Vivado echo, logging, etc.

4. Add batch insert ila.

### CODE REVIEW

File: non_project_build.tcl [hunk 1]
```diff
@@ -9,26 +9,33 @@ file mkdir $out_dir
...
 set MAX_NET_PATH_NUM 1000
 # set MAX_LOGIC_LEVEL 40
 # set MIN_LOGIC_LEVEL 4
 
-# STEP#2: setup design sources and constraints
+# STEP#2: setup design sources, ips and constraints
 #
 read_verilog [ glob $rtl_dir/*.v ]
 read_xdc [ glob $xdc_dir/*.xdc ]
+foreach f [ glob $ips_dir/*.tcl ] {
```
Comment (pwang7) [inline non_project_build.tcl]: avoid single letter as variable names

File: non_project_build.tcl [hunk 2]
```diff
@@ -78,9 +85,11 @@ if { $synth_only } {
     exit
 }
 
-# STEP#4: run logic optimization, placement and physical logic optimization,
+# STEP#4: batch insert ila, run logic optimization, placement and physical logic optimization,
 # write design checkpoint, report utilization and timing estimates
 #
+source batch_insert_ila.tcl
+batch_insert_ila 1048576
```
Comment (pwang7) [inline non_project_build.tcl]: assign literal numbers to a const

File: non_project_build.tcl [hunk 3]
```diff
@@ -78,9 +85,11 @@ if { $synth_only } {
     exit
 }
 
-# STEP#4: run logic optimization, placement and physical logic optimization,
+# STEP#4: batch insert ila, run logic optimization, placement and physical logic optimization,
 # write design checkpoint, report utilization and timing estimates
 #
+source batch_insert_ila.tcl
```
Comment (pwang7) [inline non_project_build.tcl]: add a flag to control adding ILA or not

**Review by pwang7 — CHANGES_REQUESTED** [review]

File: non_project_build.tcl (hunk 3 above)
Comment (kazutoiris) [reply non_project_build.tcl]: `batch insert ila` is disabled by default now.

File: non_project_build.tcl (hunk 2 above)
Comment (kazutoiris) [reply non_project_build.tcl]: 1048576 indicates the maximum depth.

File: non_project_build.tcl (hunk 1 above)
Comment (kazutoiris) [reply non_project_build.tcl]: fixed in the latest commit.

**Pull request merged**